stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/sql/SQLDatasource.java

```diff
@@ -61,7 +61,12 @@ public SQLDatasource() {}
 
     public boolean init(Struct options, String url, String dbType, String hostOrPath, int port, String username,
             String password, String dbName) {
-        buildDataSource(options, url, dbType, hostOrPath, dbName, port, username, password);
+        return init(options, url, dbType, hostOrPath, port, username, password, dbName, "");
+    }
+
+    public boolean init(Struct options, String url, String dbType, String hostOrPath, int port, String username,
+            String password, String dbName, String dbOptions) {
+        buildDataSource(options, url, dbType, hostOrPath, dbName, port, username, password, dbOptions);
```

Comment:
Yeah can do that. However, IMHO, even though we use it only at a few places, overloading is better here, because w.r.t the use case, the generic SQL client having "dbOptions", is not a valid scenario therefore "" doesn't have a meaning there. i.e. it makes the code less readable. As far as the person who inits the SQL client is concerned, there is no dbOptions so he shouldn't have to worry about it.